src/main/groovy/edu/oregonstate/mist/idimages/resources/IdImagesResource.groovy

```diff
@@ -35,21 +37,46 @@ class IdImagesResource extends Resource {
     @Produces("image/jpg")
     @Timed
     public Response getByOSUID(@Auth AuthenticatedUser _, @PathParam('id') String id) {
+        Integer bannerPIDM = idImageDAO.getPIDM(id)
+
+        if (!bannerPIDM) {
+            return notFound().type(MediaType.APPLICATION_JSON).build()
+        }
         Blob image = idImageDAO.getByID(id)
 
+        //Return a placeholder image if a person exists but doesn't have an ID image
         if (!image) {
-            return notFound().type(MediaType.APPLICATION_JSON).build()
+            return ok(getImageStream(ImageIO.read(new File("images/defaultImage.jpg")))).build()
         }
+        //Return an ID image of a person
         try {
-            BufferedImage buffImg = ImageIO.read(image.getBinaryStream())
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream()
-            ImageIO.write(buffImg, "jpg", outputStream)
-            byte[] imageData = outputStream.toByteArray()
-            Response.ok(imageData).build()
+            Response.ok(getImageStream(ImageIO.read(image.getBinaryStream()))).build()
         } catch (Exception e) {
             logger.error("Exception while calling getIDImages", e)
             return internalServerError("Internal server error").
                     type(MediaType.APPLICATION_JSON).build()
         }
     }
+    //Resizes a BufferedImage to a width constraint and calculates the height accordingly
+    public static BufferedImage resizeWidthConstraint(BufferedImage img, int newWidth) {
```

Comment:
I'm also wondering if the two methods that you have that deal with image manipulation should be moved to their own class. This class is handling two tasks: http requests and image manipulation. I think by moving the image manipulation to its own class that exposes only public static methods it will be easier to test and maintain over time.